x-pack/solutions/observability/plugins/infra/server/plugin.ts

```diff
@@ -103,6 +107,86 @@ export class InfraServerPlugin
       : undefined;
   }
 
+  private async createDefaultDataViewsForObsRules(
+    core: CoreStart,
+    dataViews: DataViewsServerPluginStart,
+    logsShared: LogsSharedPluginStart,
+    logsDataAccess: LogsDataAccessPluginStart
+  ) {
+    const LOG_RULES_DATA_VIEW_ID = 'log_rules_data_view';
+    const METRIC_RULES_DATA_VIEW_ID = 'infra_rules_data_view';
+
+    const LOG_RULES_DATA_VIEW_NAME = 'Log Threshold Alerting Rule Source';
+    const METRIC_RULES_DATA_VIEW_NAME = 'Metric AND Inventory Threshold Alerting Rule Source';
+
+    const override = false;
+    const skipFetchFields = false;
+    const displayErrors = true;
+    const setAsDefault = false;
+
+    const savedObjectsClient = core.savedObjects.createInternalRepository();
+    const esClient = core.elasticsearch.client.asInternalUser;
+
+    const dataViewsService = await dataViews.dataViewsServiceFactory(
+      savedObjectsClient,
+      esClient,
+      undefined,
+      true
+    );
+
+    // get log indices
+    const logSourcesService =
+      logsDataAccess.services.logSourcesServiceFactory.getLogSourcesService(savedObjectsClient);
+
+    const { indices: logIndices, timestampField } = await logsShared.logViews
+      .getClient(savedObjectsClient, core.elasticsearch.client.asInternalUser, logSourcesService)
+      .getResolvedLogView(DEFAULT_LOG_VIEW);
+
+    // create default data view for Log threshold rules
+    const logDataViewExists = await dataViewsService.get(LOG_RULES_DATA_VIEW_ID).catch(() => false);
+
+    if (!logDataViewExists) {
+      await dataViewsService.createAndSave(
```

Comment:
how does this work w/ spaces & user permissions?

Reply:
good point. We need to create data views in each space which is not the case right now. Regarding user permissions, what do you mean?

Reply:
I'm not sure if data views created with the internal user are visible for non-superusers. Probably, but would verify. For spaces, maybe you can create a data view in all spaces using `*` - I'm not sure if this is supported. And I'm also not sure what the implications then are for space-specific data views. Probably good to reach out to the team that owns data views.